lib/tests/test_revset.rs

```diff
@@ -2426,6 +2437,144 @@ fn test_evaluate_expression_author() {
+            email: "email3".to_string(),
+            timestamp: timestamp3,
+        })
+        .set_committer(Signature {
+            name: "name1".to_string(),
+            email: "email1".to_string(),
+            timestamp: timestamp2.clone(),
+        })
+        .write()
+        .unwrap();
+
+    // Can find multiple matches
+    assert_eq!(
+        resolve_commit_ids(mut_repo, "author_date(after:'2023-03-25 12:00')"),
+        vec![commit3.id().clone(), commit2.id().clone()]
+    );
+    assert_eq!(
+        resolve_commit_ids(mut_repo, "author_date(before:'2023-03-25 12:00')"),
+        vec![commit1.id().clone(), root_commit.id().clone()]
+    );
+}
+
+#[test]
+fn test_evaluate_expression_committer_date() {
+    let settings = testutils::user_settings();
+    let test_repo = TestRepo::init();
+    let repo = &test_repo.repo;
+
+    let mut tx = repo.start_transaction(&settings);
+    let mut_repo = tx.mut_repo();
+
+    let timestamp1 = parse_timestamp("2023-03-25T11:30:00Z");
+    let timestamp2 = parse_timestamp("2023-03-25T12:30:00Z");
+    let timestamp3 = parse_timestamp("2023-03-25T13:30:00Z");
+
+    let root_commit = repo.store().root_commit();
+    let commit1 = create_random_commit(mut_repo, &settings)
+        .set_author(Signature {
+            name: "name1".to_string(),
+            email: "email1".to_string(),
+            timestamp: timestamp2.clone(),
+        })
+        .set_committer(Signature {
+            name: "name1".to_string(),
+            email: "email1".to_string(),
+            timestamp: timestamp1.clone(),
+        })
+        .write()
+        .unwrap();
+    let commit2 = create_random_commit(mut_repo, &settings)
+        .set_parents(vec![commit1.id().clone()])
+        .set_author(Signature {
+            name: "name2".to_string(),
+            email: "email2".to_string(),
+            timestamp: timestamp2.clone(),
+        })
+        .set_committer(Signature {
+            name: "name1".to_string(),
+            email: "email1".to_string(),
+            timestamp: timestamp2.clone(),
+        })
+        .write()
+        .unwrap();
+    let commit3 = create_random_commit(mut_repo, &settings)
+        .set_parents(vec![commit2.id().clone()])
+        .set_author(Signature {
+            name: "name3".to_string(),
+            email: "email3".to_string(),
+            timestamp: timestamp2.clone(),
+        })
+        .set_committer(Signature {
+            name: "name1".to_string(),
+            email: "email1".to_string(),
+            timestamp: timestamp3,
+        })
+        .write()
+        .unwrap();
+
+    // Can find multiple matches
+    assert_eq!(
+        resolve_commit_ids(mut_repo, "committer_date(after:'2023-03-25 12:00')"),
+        vec![commit3.id().clone(), commit2.id().clone()]
+    );
+    assert_eq!(
+        resolve_commit_ids(mut_repo, "committer_date(before:'2023-03-25 12:00')"),
+        vec![commit1.id().clone(), root_commit.id().clone()]
+    );
+}
+
```

Comment:
Can you add a few `Local` timezone tests to exercise the production code? It could be a CLI tests.

Reply:
Added a test `test_revset_committer_date_with_timezone` that creates some commits and verifies that, after changing the TZ env var, the matching revisions changes.

It would probably be nice to verify that it understands DST shifts, but I'm not sure how to write that test because I'd need to control the current time. For example, if it's 2024-06-01 and UTC-4 in New York and you enter `committer_date(after:"2024-01-01")`, it _should_ know that the offset at that time was UTC-5 so that means `2024-01-01T05:00:00Z`.

Reply:
Maybe we can test with two or three different timezones whose DST ranges don't overlap? I don't know how hard it would be, though.

BTW, chrono appears to support `TZ` variable with DST range:
https://github.com/chronotope/chrono/blob/43579a5304f9433ce42614e200b554ea8e0361cc/src/offset/local/tz_info/rule.rs#L20
It'll probably help stabilize the tests.

Reply:
I added another check that I get the expected results from an Australian time zone. I feel pretty confident that it's behaving as I wanted.